lib/cyclops/RPC_defs.h

```diff
@@ -1,13 +1,80 @@
-        [7:6] Channel
-        [5:0] Command
-    ~~~~~~~~~~~~~~~~~~~~
+    
+    @page rpc-desc RPC Format Description
+
+    @tableofcontents
+
+    @section rpc-overview Overview
+    There are two kinds of Serial packets, single byte and multi byte.
+    The size of a packet is determined by MSB of header and the ``command`` field.
+
+    @section S-header-desc Single Byte Packets
+    Single byte Headers are distinguished from other headers by the MSB bit. For
+    Single byte headers, this bit is always set
+    
+    Field       | Bits  | Description
+    ----------- | ----- | -----------
+    Reserved    | [7]   | Always ``1``.
+    ``channel`` | [6:3] | Determines if "command" is applied on the Channel @par 1: Command is appplied on Channel
+    ``command`` | [2:0] | The command field
+    
+    @subsection S-cmd-desc Command descriptions
+    
+    command[3:0] | Name     | Effect
+    ------------ | -------  | -------------
+    ``0000``     | start    | Launch mainloop
+    ``0001``     | stop     | Pause mainloop
+    ``0010``     | reset    | Reset all sources. @attention The system is *not* reset to _initial configuration_!
+    ``0011``     | swap     | Swap the Cyclops instances of the 2 high ``channel`` bits.
+    ``1111``     | identity | Send device description
+    
+    @section M-header-desc Multi Byte Packets
+    Packet is formed by concatenating the header with argument bytes.
+
+    Field       | Bits  | Description
+    ----------- | ----- | -----------
+    Reserved    | [7]   | Always ``0``.
+    ``channel`` | [6:5] | Command is appplied on Channel ``channel[1:0]``
+    ``command`` | [4:0] | The command field
+
+    @subsection M-cmd-desc Command descriptions
+
+    command[4:0]  | Name               | Size(Bytes) | Effect
+    ------------- | ------------------ | ----------- | --------
+    ``00000``     | change_source_l    | 2           | Changes Source instance to the one reffered by @ref src-id-sec. Mode is set to ``LOOPBACK``.
+    ``00001``     | change_source_o    | 2           | Changes Source instance to the one reffered by @ref src-id-sec. Mode is set to ``ONE_SHOT``.
+    ``00010``     | change_source_n    | 3           | Changes Source instance to the one reffered by @ref src-id-sec. Mode is set to ``N_SHOT``. \f$N\f$ is set to @ref shot_cycle "shot_cycle"
+    ``00011``     | change_time_period | 5           | Set time period of updates @attention Works only if Source::holdTime is a constant!
+    ``00101``     | time_factor        | 5           | Scale Source::holdTime values by this factor. \f$\in (0, \infty)\f$.
+    ``00111``     | voltage_offset     | 3           | Add this DC offset level to Source::getVoltage values. \f$\in (0, \infty)\f$.
+    ``01000``     | square_on_time     | 5           | Set squareSource::onTime.
+    ``01001``     | square_off_time    | 5           | Set squareSource::offTime.
+
+    @note       Unlike providing a scaling factor for time, we don't provide a
```

Comment:
Well -- what if somebody wants to scale the power depending on a neural response. They can't do this with the knob. We must therefore have a way to adjust the amplitude via this interface.